thewalrus/tests/test_operations.py

```diff
@@ -0,0 +1,166 @@
+            [-0.09544978 + 1.78968334e-01j, 0.29708743 + 4.95145724e-01j, 0.31873657 + 0.00000000e00j, -0.29777767 + 4.96296112e-01j, -0.18306015 - 3.43237787e-01j],
+            [-0.06819609 + 3.44424719e-03j, -0.14658716 + 2.74850926e-01j, 0.29777767 + 4.96296112e-01j, 0.12389162 + 1.10385981e-17j, -0.27646677 + 4.60777945e-01j],
+            [-0.01109048 - 1.65323865e-02j, -0.12479885 + 6.30297236e-03j, -0.18306015 + 3.43237787e-01j, 0.27646677 + 4.60777945e-01j, -0.03277289 + 1.88440656e-17j],
+        ]
+    )
+    T = operations.fock_tensor(S, alphas, cutoff)
+    assert np.allclose(T, expected, atol=tol, rtol=0)
+
+
+def test_single_mode_squeezing(tol):
+    """Tests the correct construction of the single mode squeezing operation"""
+    nmodes = 1
+    r = 1.0
+    cutoff = 5
+    S = squeezing(r, 0.0)
+    alphas = np.zeros([nmodes])
+    # This data is obtained by using qutip
+    # np.array(squeeze(40,r).data.todense())[0:5,0:5]
+    expected = np.array(
+        [
+            [0.80501818 + 0.0j, 0.0 + 0.0j, 0.43352515 + 0.0j, 0.0 + 0.0j, 0.2859358 + 0.0j],
+            [0.0 + 0.0j, 0.52169547 + 0.0j, 0.0 + 0.0j, 0.48661591 + 0.0j, 0.0 + 0.0j],
+            [-0.43352515 + 0.0j, 0.0 + 0.0j, 0.10462138 + 0.0j, 0.0 + 0.0j, 0.29199268 + 0.0j],
+            [0.0 + 0.0j, -0.48661591 + 0.0j, 0.0 + 0.0j, -0.23479643 + 0.0j, 0.0 + 0.0j],
+            [0.2859358 + 0.0j, 0.0 + 0.0j, -0.29199268 + 0.0j, 0.0 + 0.0j, -0.34474749 + 0.0j],
+        ]
+    )
+    T = operations.fock_tensor(S, alphas, cutoff)
+    assert np.allclose(T, expected, atol=tol, rtol=0)
+
+
+def test_single_mode_displacement_squeezing(tol):
+    """Tests the correct construction of the single mode squeezing operation followed by the single mode displacement operation"""
+    nmodes = 1
+    r = 1.0
+    cutoff = 5
+    S = squeezing(r, 0.0)
+    alphas = (0.5 + 0.4 * 1j) * np.ones([nmodes])
+    # This data is obtained by using qutip
+    # np.array((displace(40,alpha)*squeeze(40,r)).data.todense())[0:5,0:5]
+    expected = np.array(
+        [
+            [0.6263739 + 0.09615331j, -0.22788717 + 0.13121343j, 0.36548296 - 0.0200537j, -0.20708137 + 0.14004403j, 0.25645667 - 0.06275564j],
+            [0.5425389 + 0.14442404j, 0.19268911 + 0.15615312j, 0.11497303 + 0.13744549j, 0.21448948 + 0.08109308j, -0.03652914 + 0.15069359j],
+            [-0.00915607 + 0.07475267j, 0.48081922 + 0.10576742j, -0.00961086 + 0.20535144j, 0.33089303 + 0.09864247j, 0.02555522 + 0.19950786j],
+            [-0.34614367 - 0.05229875j, 0.11543956 + 0.01112537j, 0.16672961 + 0.07439407j, 0.02323121 + 0.15103267j, 0.27233637 + 0.08297028j],
+            [-0.14390852 - 0.08884069j, -0.37898007 - 0.07630228j, 0.12911863 - 0.08963054j, -0.12164023 + 0.04431394j, 0.1141808 + 0.01581529j],
+        ]
+    )
+    T = operations.fock_tensor(S, alphas, cutoff)
+    assert np.allclose(T, expected, atol=tol, rtol=0)
```

Comment:
Should we also have a test for the S2 gate and beamsplitter?

Reply:
For beamsplitter there are several, since interferometer supersedes beamsplitter.
For S2 it would be nice but not sure where to find reference values. There however selection rules that can be checked. Will add one for those rules.